File: 프로그래머스/lv1/12906. 같은 숫자는 싫어/같은 숫자는 싫어.py
def solution(arr):
    answer = []
    lst = list()

    lst.append(arr[0])
    answer.append(arr[0])

    i = 1
    while i < len(arr):
        tmp = lst.pop(0)

        if tmp != arr[i]:
            answer.append(arr[i])

        lst.append(arr[i])
        i += 1

    return answer

File: 프로그래머스/lv1/12906. 같은 숫자는 싫어/같은 숫자는 싫어.java
import java.util.*;

public class Solution {
    public Stack<Integer> solution(int []arr) {
        Stack<Integer> stack = new Stack<>();
        int i;
        
        stack.push(arr[0]);
        for(i = 1; i < arr.length; i++){
            int tmp = stack.pop();
            if(tmp != arr[i]){
                stack.push(tmp);
            }
            stack.push(arr[i]);
        }
        
        return stack;
    }
}

